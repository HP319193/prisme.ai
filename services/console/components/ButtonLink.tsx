import Link from "next/link"
import { ButtonHTMLAttributes, FC } from "react"

interface ButtonLinkProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  href?: string;
}
export const ButtonLink: FC<ButtonLinkProps> = ({ children, href, ...props }) => {

  return <button style={{ background: 'none', border: 0, cursor: 'pointer', fontSize: 'inherit', ...props.style }} {...props}>
    {href && <Link href={href}><>{children}</></Link>}
    {!href && children}
  </button>
}

export default ButtonLink
