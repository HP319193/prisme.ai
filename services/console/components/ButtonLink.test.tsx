import ButtonLink from "./ButtonLink";
import renderer from "react-test-renderer";

jest.mock("next/link", () => {
  const Link = ({ children }: any) => <div className="Link">{children}</div>
  return Link
})
it("should render without link", () => {
  const root = renderer.create(<ButtonLink>Click</ButtonLink>);
  expect(root.toJSON()).toMatchSnapshot();
});

it("should render with link", () => {
  const root = renderer.create(<ButtonLink href="/foo">Click</ButtonLink>);
  expect(root.toJSON()).toMatchSnapshot();
});
