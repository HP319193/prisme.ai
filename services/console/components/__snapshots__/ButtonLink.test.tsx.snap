// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with link 1`] = `
<button
  style={
    Object {
      "background": "none",
      "border": 0,
      "cursor": "pointer",
      "fontSize": "inherit",
    }
  }
>
  <div
    className="Link"
  >
    Click
  </div>
</button>
`;

exports[`should render without link 1`] = `
<button
  style={
    Object {
      "background": "none",
      "border": 0,
      "cursor": "pointer",
      "fontSize": "inherit",
    }
  }
>
  Click
</button>
`;
